src/gui/search/searchwidget.cpp

```diff
@@ -306,173 +303,84 @@ void SearchWidget::on_searchButton_clicked()
+
+        if (m_mainWindow->isNotificationsEnabled() && (m_mainWindow->currentTabWidget() != this)) {
+            if (m_activeSearchTab->status() == SearchTab::Status::Error)
+                m_mainWindow->showNotificationBaloon(tr("Search Engine"), tr("Search has failed"));
+            else
+                m_mainWindow->showNotificationBaloon(tr("Search Engine"), tr("Search has finished"));
+        }
 
-        curModel->setData(curModel->index(row, SearchSortModel::DL_LINK), result.fileUrl); // download URL
-        curModel->setData(curModel->index(row, SearchSortModel::NAME), result.fileName); // Name
-        curModel->setData(curModel->index(row, SearchSortModel::SIZE), result.fileSize); // Size
-        curModel->setData(curModel->index(row, SearchSortModel::SEEDS), result.nbSeeders); // Seeders
-        curModel->setData(curModel->index(row, SearchSortModel::LEECHES), result.nbLeechers); // Leechers
-        curModel->setData(curModel->index(row, SearchSortModel::ENGINE_URL), result.siteUrl); // Search site URL
-        curModel->setData(curModel->index(row, SearchSortModel::DESC_LINK), result.descrLink); // Description Link
+        m_activeSearchTab = nullptr;
+        m_ui->searchButton->setText(tr("Search"));
     }
-
-    m_noSearchResults = false;
-    m_activeSearchTab->updateResultsCount();
-
-    // Enable clear & download buttons
-    m_ui->downloadButton->setEnabled(true);
-    m_ui->goToDescBtn->setEnabled(true);
-    m_ui->copyURLBtn->setEnabled(true);
 }
 
 void SearchWidget::closeTab(int index)
 {
-    // Search is run for active tab so if user decided to close it, then stop search
-    if (!m_activeSearchTab.isNull() && index == m_ui->tabWidget->indexOf(m_activeSearchTab)) {
-        qDebug("Closed active search Tab");
-        if (m_searchEngine->isActive())
-            m_searchEngine->cancelSearch();
-        m_activeSearchTab = 0;
-    }
-
-    delete m_allTabs.takeAt(index);
+    SearchTab *tab = m_allTabs.takeAt(index);
+    if (tab == m_activeSearchTab)
+        m_ui->searchButton->setText(tr("Search"));
 
-    if (!m_allTabs.size()) {
-        m_ui->downloadButton->setEnabled(false);
-        m_ui->goToDescBtn->setEnabled(false);
-        m_ui->copyURLBtn->setEnabled(false);
-    }
+    delete tab;
 }
 
 // Download selected items in search results list
 void SearchWidget::on_downloadButton_clicked()
 {
-    //QModelIndexList selectedIndexes = currentSearchTab->getCurrentTreeView()->selectionModel()->selectedIndexes();
-    QModelIndexList selectedIndexes =
-        m_allTabs.at(m_ui->tabWidget->currentIndex())->getCurrentTreeView()->selectionModel()->selectedIndexes();
-    foreach (const QModelIndex &index, selectedIndexes) {
-        if (index.column() == SearchSortModel::NAME)
-            m_allTabs.at(m_ui->tabWidget->currentIndex())->downloadItem(index);
-    }
+    m_allTabs.at(m_ui->tabWidget->currentIndex())->downloadTorrents();
```

Comment:
now you can only download 1 torrent at a time right? will users like it?

Reply:
or handled in `SearchTab::downloadTorrents()`?

Reply:
>now you can only download 1 torrent at a time (and also similar operations below) right?

No. I call `SearchTab::downloadTorrents()` here that downloads all selected torrents.